# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/ipc/WebrtcTCPSocket.cpp
# Commit: da41f0fab427
# Full Hash: da41f0fab427b199f1a0c6156467a0ad9a9e6891
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Make proxy config parameters optional in WebrtcTCPSocket to get the NAT simulator working again. r=mjf,mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45104
# ==============================================================================

diff -r b83124cf3a89 -r da41f0fab427 media/mtransport/ipc/WebrtcTCPSocket.cpp
--- a/media/mtransport/ipc/WebrtcTCPSocket.cpp	Tue Sep 17 18:17:24 2019 +0000
+++ b/media/mtransport/ipc/WebrtcTCPSocket.cpp	Tue Sep 17 18:17:37 2019 +0000
@@ -136,12 +136,11 @@
   InvokeOnClose(aReason);
 }
 
-nsresult WebrtcTCPSocket::Open(const nsCString& aHost, const int& aPort,
-                               const nsCString& aLocalAddress,
-                               const int& aLocalPort, bool aUseTls,
-                               const net::LoadInfoArgs& aArgs,
-                               const nsCString& aAlpn,
-                               NrSocketProxyConfig::ProxyPolicy aProxyPolicy) {
+nsresult WebrtcTCPSocket::Open(
+    const nsCString& aHost, const int& aPort, const nsCString& aLocalAddress,
+    const int& aLocalPort, bool aUseTls, const Maybe<net::LoadInfoArgs>& aArgs,
+    const Maybe<nsCString>& aAlpn,
+    const Maybe<NrSocketProxyConfig::ProxyPolicy>& aProxyPolicy) {
   LOG(("WebrtcTCPSocket::Open %p\n", this));
 
   if (NS_WARN_IF(mOpened)) {
@@ -165,21 +164,23 @@
     return NS_ERROR_FAILURE;
   }
 
-  mLoadInfoArgs = aArgs;
-  mAlpn = aAlpn;
   mTls = aUseTls;
-  if (aProxyPolicy == NrSocketProxyConfig::kForceProxy) {
-    mForceProxy = true;
-  }
   mLocalAddress = aLocalAddress;
   mLocalPort = aLocalPort;
 
-  if (aProxyPolicy == NrSocketProxyConfig::kDisableProxy) {
-    rv = OpenWithoutHttpProxy(nullptr);
+  if (aProxyPolicy.isSome() &&
+      *aProxyPolicy != NrSocketProxyConfig::kDisableProxy) {
+    mLoadInfoArgs = *aArgs;
+    mAlpn = *aAlpn;
+    if (*aProxyPolicy == NrSocketProxyConfig::kForceProxy) {
+      mForceProxy = true;
+    }
+
+    // We need to figure out whether a proxy needs to be used for mURI before
+    // we can start on establishing a connection.
+    rv = DoProxyConfigLookup();
   } else {
-    // We need to figure out whether a proxy needs to be used for mURI before we
-    // can start on establishing a connection.
-    rv = DoProxyConfigLookup();
+    rv = OpenWithoutHttpProxy(nullptr);
   }
 
   if (NS_WARN_IF(NS_FAILED(rv))) {